test/integration/fixtures/fixtures.go

```diff
@@ -239,14 +240,19 @@ func GetExpectedRbac(o *OverrideRbac) *ExpectedRbac {
 
 		roleFQDN := string(signedDomain.Domain.Roles[i].Name)
 		roleName := strings.TrimPrefix(roleFQDN, fmt.Sprintf("%s:role.", domainName))
-		sr := common.NewConfig(model.ServiceRole.Type, ns, roleName, srSpec)
+		sr := common.NewConfig(collections.IstioRbacV1Alpha1Serviceroles, ns, roleName, srSpec)
 		modelConfig = append(modelConfig, sr)
 
-		if len(signedDomain.Domain.Roles[i].RoleMembers) > 0 {
-			srbSpec.RoleRef.Name = sr.Name
-			srb := common.NewConfig(model.ServiceRoleBinding.Type, ns, roleName, srbSpec)
-			modelConfig = append(modelConfig, srb)
+		// Every Role has SRB with Spiffe URI
+		if len(signedDomain.Domain.Roles[i].RoleMembers) == 0 {
+			srbSpec.Subjects = []*v1alpha1.Subject{}
 		}
+		srbSpec.Subjects = append(srbSpec.Subjects, &v1alpha1.Subject{
+			User: fmt.Sprintf("%s/ra/%s", signedDomain.Domain.Name, sr.Name),
```

Comment:
we should ideally also put this in `getDefaultServiceRoleBinding` since all SRB should have role cert spiffe added by default now

Reply:
`getDefaultServiceRoleBinding` does not have any namespace info so moved adding spiffe uri here where namespace info is available.